src/main/java/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepository.java

```diff
@@ -333,37 +304,54 @@ private void assertPathExists(String path) {
     }
 
     public ScmPosition currentPosition() {
+        String revision = getRevision();
+        String branchName = branchName();
+        boolean isClean = !checkUncommittedChanges();
+        return new ScmPosition(revision, branchName, isClean);
+    }
+
+    private String getRevision() {
         try {
             String revision = "";
             if (hasCommits()) {
                 ObjectId head = head();
                 revision = head.name();
             }
-
-            boolean isClean = !checkUncommittedChanges();
-
-            String branchName = branchName();
-            return new ScmPosition(revision, branchName, isClean);
+            return revision;
         } catch (IOException e) {
             throw new ScmException(e);
         }
     }
 
+    private String branchName() {
+        return branchNameFromEnvVariable().orElseGet(this::branchNameFromGit);
+    }
+
+    /**
+     * @return head ref or source branch of the pull request in a workflow run
+     */
+    private Optional<String> branchNameFromEnvVariable() {
+        return Optional.ofNullable(System.getenv("GITHUB_HEAD_REF"));
```

Comment:
We need to handle multiple trigger types here (based on GITHUB_EVENT_NAME):
`push` -> GITHUB_REF_NAME
`pull-request`, `pull_request_target` -> GITHUB_HEAD_REF

We should only apply this logic where GITHUB_REF_TYPE is `branch` probably too.

Sources: 
https://docs.github.com/en/actions/learn-github-actions/variables
https://docs.github.com/en/actions/learn-github-actions/variables